### ISSUE TITLE
What is the purpose of `share_planes`?

### ISSUE THREAD

**Issue by blurgyy** [POSTECH-CVLab/point-transformer#29]
Hello, thank you for uploading this!

I'm not sure what does `share_planes` do in the recently added code from the paper authors.  It seem that the variable `share_planes` was originally defined here (where `fpn_planes` and `fpnhead_planes` were not used anywhere else):

https://github.com/POSTECH-CVLab/point-transformer/blob/7009caf16e798ea3aa0e35166b82094a2ad5417b/model/pointtransformer/pointtransformer_seg.py#L129

Later, in `PointTransformerLayer`'s `__init__()` and `forward()` functions, `share_planes` is mostly used as a denominator, to adjust the intermediate dimensions between layers, like here (in function `PointTransformerLayer.__init__()`):

https://github.com/POSTECH-CVLab/point-transformer/blob/7009caf16e798ea3aa0e35166b82094a2ad5417b/model/pointtransformer/pointtransformer_seg.py#L18-L21

And here (in function `PointTransformerLayer.forward()`):

https://github.com/POSTECH-CVLab/point-transformer/blob/7009caf16e798ea3aa0e35166b82094a2ad5417b/model/pointtransformer/pointtransformer_seg.py#L34-L35

I failed to infer what does `share_planes` actually mean.  My question here is, when a network layer is constructed by

```python
nn.Linear(mid_planes, mid_planes // share_planes)
```

- Does it only mean that the layer will project a `mid_planes`-dimensional representation onto a `mid_planes//share_planes`-dimensional space?
- Does the word _share_ in the variable name `share_planes` have any meaning?

Also, the unused `fpn_planes` and `fpnhead_planes` variables are confusing.  Thank you for taking the time to read this, any response would be greatly appreciated.

**Comment by blurgyy** [comment]
So since nobody answers this issue, here is my guess (after a few weeks I opened this issue): according to the last aggregating step in the `forward` method of `PointTransformerLayer`:

https://github.com/POSTECH-CVLab/point-transformer/blob/8d2a38998f1ed8cd6d03fe1b671440724aa269c8/model/pointtransformer/pointtransformer_seg.py#L34-L35

`share_planes` specifies how many planes will share a same weight in this step.

---

Closing.  Anyone with a different opinion is welcome to reopen this issue/discuss with me.

**Comment by wangbxj1234** [comment]
And c=6,6//8=0, right?  what's the output dim of the linear layer ?

**Comment by blurgyy** [comment]
Hi @wangbxj1234, actually the value of `c` here can only be one of `[32, 64, 128, 256, 512]`, according to this line:

https://github.com/POSTECH-CVLab/point-transformer/blob/8d2a38998f1ed8cd6d03fe1b671440724aa269c8/model/pointtransformer/pointtransformer_seg.py#L128

The aggregation step mentioned in my last comment is the final step of the `PointTransformerLayer`, and this layer is only applied after the original point feature dimension (which is 6) is transformed into 32 and other 8's multiples.

You can refer to Figure.3 in the original paper.  Please point out if I'm missing anything.

**Comment by jinghere11** [comment]
> Hi @wangbxj1234, actually the value of `c` here can only be one of `[32, 64, 128, 256, 512]`, according to this line:
> 
> https://github.com/POSTECH-CVLab/point-transformer/blob/8d2a38998f1ed8cd6d03fe1b671440724aa269c8/model/pointtransformer/pointtransformer_seg.py#L128
> 
> The aggregation step mentioned in my last comment is the final step of the `PointTransformerLayer`, and this layer is only applied after the original point feature dimension (which is 6) is transformed into 32 and other 8's multiples.
> 
> You can refer to Figure.3 in the original paper. Please point out if I'm missing anything.

        self.in_planes = planes * block.expansion
